data/scripting/win_conditions/defeat_hq.lua

```diff
@@ -0,0 +1,60 @@
+-- =======================================================================
+--                         Defeat all Win condition
+-- =======================================================================
+
+include "scripting/coroutine.lua" -- for sleep
+include "scripting/win_conditions/win_condition_functions.lua"
+
+set_textdomain("win_conditions")
+
+include "scripting/win_conditions/win_condition_texts.lua"
+
+local wc_name = "HQ Hunter"
+-- This needs to be exactly like wc_name, but localized, because wc_name
+-- will be used as the key to fetch the translation in C++
+local wc_descname = _("HQ Hunter")
+local wc_version = 1
+local wc_desc = _ "The tribe or team that can destroy all other headquarters wins the game!"
+return {
+    name = wc_name,
+    description = wc_desc,
+    peaceful_mode_allowed = false,
+    func = function()
+        local plrs = wl.Game().players
+
+        -- set the objective with the game type for all players
+        broadcast_objective("win_condition", wc_descname, wc_desc)
+
+        -- Iterate all players, if one is defeated, remove him
+        -- from the list, send him a defeated message and give him full vision
+        repeat
+            sleep(5000)
+
+            for idx,p in ipairs(plrs) do
+                if (#p:get_buildings("barbarians_headquarters") + #p:get_buildings("atlanteans_headquarters") + #p:get_buildings("empire_headquarters") + #p:get_buildings("frisians_headquarters")) == 0 then
+                    for idx,b in ipairs(p:get_buildings("barbarians_warehouse")) do
+                        b:destroy()
+                    end
+                    for idx,b in ipairs(p:get_buildings("atlanteans_warehouse")) do
+                        b:destroy()
+                    end
+                    for idx,b in ipairs(p:get_buildings("empire_warehouse")) do
+                        b:destroy()
+                    end
+                    for idx,b in ipairs(p:get_buildings("frisians_warehouse")) do
+                        b:destroy()
+                    end
```

Comment:
Suggestion: To reduce the amount of typing, you can add `include "scripting/table.lua"`and then use
~~~~
for idx,b in ipairs(array_combine(
   p:get_buildings("barbarians_warehouse"),
   p:get_buildings("empire_warehouse"),
   …etc… )) do
~~~~

Reply:
Thanks. Implemented in 0cbda9d